src/back/msl/writer.rs

```diff
@@ -1322,60 +1318,64 @@ impl<W: Write> Writer<W> {
         Ok(())
     }
 
-    fn write_inline_sampler(
+    fn put_inline_sampler_properties(
         &mut self,
+        level: Level,
         sampler: &sm::InlineSampler,
-        stage: crate::ShaderStage,
-        handle: Handle<crate::GlobalVariable>,
     ) -> Result<(), Error> {
-        write!(
-            self.out,
-            "constexpr sampler ism_{}_{:?}(",
-            handle.index(),
-            stage
-        )?;
         for (&letter, address) in ['s', 't', 'r'].iter().zip(sampler.address.iter()) {
-            write!(
+            writeln!(
                 self.out,
-                "{}{}_address::{},",
-                INDENT,
+                "{}{}::{}_address::{},",
+                level,
+                NAMESPACE,
                 letter,
-                address.as_str()
+                address.as_str(),
             )?;
         }
-        write!(
+        writeln!(
             self.out,
-            "{}mag_filter::{},",
-            INDENT,
-            sampler.mag_filter.as_str()
+            "{}{}::mag_filter::{},",
+            level,
+            NAMESPACE,
+            sampler.mag_filter.as_str(),
         )?;
-        write!(
+        writeln!(
             self.out,
-            "{}min_filter::{},",
-            INDENT,
-            sampler.min_filter.as_str()
+            "{}{}::min_filter::{},",
+            level,
+            NAMESPACE,
+            sampler.min_filter.as_str(),
         )?;
         if let Some(filter) = sampler.mip_filter {
-            write!(self.out, "{}mip_filter::{},", INDENT, filter.as_str())?;
+            writeln!(
+                self.out,
+                "{}{}::mip_filter::{},",
+                level,
+                NAMESPACE,
+                filter.as_str(),
+            )?;
         }
         // avoid setting it on platforms that don't support it
         if sampler.border_color != sm::BorderColor::TransparentBlack {
-            write!(
+            writeln!(
                 self.out,
-                "{}border_color::{},",
-                INDENT,
-                sampler.border_color.as_str()
+                "{}{}::border_color::{},",
+                level,
+                NAMESPACE,
+                sampler.border_color.as_str(),
             )?;
         }
         if sampler.compare_func != sm::CompareFunc::Never {
-            write!(
+            writeln!(
                 self.out,
-                "{}compare_func::{},",
-                INDENT,
-                sampler.compare_func.as_str()
+                "{}{}::compare_func::{},",
+                level,
+                NAMESPACE,
+                sampler.compare_func.as_str(),
             )?;
         }
-        writeln!(self.out, "{}coord::normalized);", INDENT)?;
+        writeln!(self.out, "{}{}::coord::normalized", NAMESPACE, level)?;
```

Comment:
`NAMESPACE` and `level` should be switched here

Reply:
just noticed this as well :)